src/injectHighlightWrappers.ts

```diff
@@ -235,7 +235,10 @@ function refineRangeBoundaries(range: Range) {
     if (range.endOffset < endContainer.nodeValue!.length) {
       (endContainer as Text).splitText(range.endOffset);
     }
-  } else if (range.endOffset > 0) {
+  } else if (range.endOffset > 0 && !dom(endContainer).matches('iframe')) {
+    // ranges ending in an iframe (observed in firefox) should not update their endContainer
+    // otherwise a child of the iframe such as the <!-- no-selfclose --> comment will be assigned
+    // this will prevent the highlight from being generated correctly in highlightRange()
```

Comment:
@TomWoodward latest findings: a highlight starting/ending on an iframe with any child (comment, script tag, text, or otherwise) causes an error in [highlightRange](https://github.com/openstax/highlighter/blob/master/src/injectHighlightWrappers.ts#L155), the logic of which I am still trying to wrap my head around. But from what I can tell the range endContainer is reassigned to the iframe child and then that child is passed into `highlightRange` which traverses the tree from the startContainer looking for a common ancestor to highlight. During that loop the iframe child is never found so the highlight is not created correctly.

If you remove the comment from the iframe then that function runs without error, but `hasBlockContent` returns false, which prevents the right classes from being added to the highlight to show color around the iframe. I haven't looked into why that is the case yet. 

Another thing to investigate: why, with or without a child node, the iframe highlight is always displayed correctly in MH, including on Chrome.